src/type/definition.js

```diff
@@ -506,6 +506,11 @@ function defineFieldMap<TSource, TContext>(
       `${type.name}.${fieldName} field type must be Output Type but ` +
       `got: ${String(field.type)}.`
     );
+    invariant(
+      isValidResolver(field.resolve),
```

Comment:
I tried that, but the linter treats == as an error. I can update the linter to allow it if you feel strongly that == is more expressive.